sdk/tests/utests/Job_tests.cpp

```diff
@@ -0,0 +1,279 @@
+    bool                    isExecuting     = false;
+    bool                    finishExecution = false;
+    std::mutex              executorMutex;
+    std::condition_variable executorCV;
+    Job                     job([&] {
+        {
+            std::lock_guard guard{executorMutex};
+            isExecuting = true;
+            executorCV.notify_one();
+        }
+        {
+            std::unique_lock lock(executorMutex);
+            executorCV.wait(lock, [&] { return finishExecution; });
+        }
+    });
+    std::thread             executor([&] {
+        job.execute();
+        {
+            std::lock_guard guard{executorMutex};
+            isExecuting = false;
+            executorCV.notify_one();
+        }
+    });
+
+    // make sure job is in execution
+    {
+        std::unique_lock lock(executorMutex);
+        executorCV.wait(lock, [&] { return isExecuting; });
+    }
+
+    // setup the scavenger (waiting for termination of the job)
+    bool                    startedWaiting  = false;
+    bool                    finishedWaiting = false;
+    std::mutex              scavengerMutex;
+    std::condition_variable scavengerCV;
+    std::thread             scavenger([&] {
+        {
+            std::lock_guard guard{scavengerMutex};
+            startedWaiting = true;
+        }
+        scavengerCV.notify_one();
+        job.waitForTermination();
+        {
+            std::lock_guard guard{scavengerMutex};
+            finishedWaiting = true;
+        }
+        scavengerCV.notify_one();
+    });
+
+    // make sure scavenger is waiting
+    {
+        std::unique_lock lock{scavengerMutex};
+        scavengerCV.wait(lock, [&] { return startedWaiting; });
+        std::this_thread::sleep_for(2ms);
+    }
+
+    // Notify executor to finish
+    {
+        std::lock_guard guard{executorMutex};
+        finishExecution = true;
+        executorCV.notify_one();
+    }
+
+    // Wait till executor is done
+    {
+        std::unique_lock lock{executorMutex};
+        executorCV.wait(lock, [&] { return !isExecuting; });
+    }
+
+    // make sure scavenger is done
+    {
+        std::unique_lock lock{scavengerMutex};
+        scavengerCV.wait(lock, [&] { return finishedWaiting; });
+        EXPECT_TRUE(finishedWaiting);
+    }
+
+    executor.join();
+    scavenger.join();
+}
+
+TEST(Test_RecurringJob, callCtor_none_noFailure) {
+    EXPECT_NO_THROW(RecurringJob([] {}));
+}
+
+TEST(Test_RecurringJob, create_firstCall_noFailureAndNoNullPtr) {
+    JobPtr_t job;
+    EXPECT_NO_THROW(job = RecurringJob::create([] {}));
+    EXPECT_TRUE(job);
+}
+
+TEST(Test_RecurringJob, create_jobsCreatedBefore_noFailureAndReturnsNewValidPtr) {
+    JobPtr_t job1 = RecurringJob::create([] {});
+
+    JobPtr_t job2;
+    EXPECT_NO_THROW(job2 = RecurringJob::create([] {}));
+    EXPECT_TRUE(job2);
+    EXPECT_NE(job1.get(), job2.get());
+}
+
+TEST(Test_RecurringJob, shallRecur_freshlyCreated_returnsTrue) {
```

Comment:
```suggestion
TEST(Test_RecurringJob, shallRecur_afterConstruction_returnsTrue) {
```

Reply:
Preferring "freshlyCreated" (or freshlyConstructed) because "after construction" can be any arbitrary time after construction. I would be more precise to say "immediatlyAfterConstruction" ...